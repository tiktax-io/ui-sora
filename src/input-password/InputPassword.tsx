import React, { forwardRef, LegacyRef, memo, useState } from 'react'
import Icon from '../icon/Icon'
import { IconProps } from '../icon/Icon.types'
import { black, danger, success } from '../_styles'

import { isInvalidStyle, isValidStyle, useStyles } from './InputPassword.styles'
import { InputPasswordProps } from './InputPassword.types'

/******************************************************************************
 * Functional Component that renders a text input completed with label and
 * optional validation styles and message.
 * @param {string} [dataTestId] id of the container element to be
 * used on tests as a way to query the component.
 * @param {string} feedbackMessage validation message to display below the
 * input field.
 * @param {boolean} isInvalid true if the input will display an error status.
 * @param {boolean} isValid true if the input will display a success status.
 * @param {string} label label title of the input.
 * @param {() => unknown} onChange function to trigger when input value changes.
 * @param {boolean} passwordIsHidden true by default. False to show password.
 * @param {string} placeholder string used as placeholder for user instructions.
 * @param {string} value the string value of the input for controlled forms.
 * @param {number} width integer representing the width in pixels.
 * @returns {ReactElement} Div containing text input, optionally label above,
 * and feedback message below.
 *****************************************************************************/
const InputPassword = memo(forwardRef(({
  dataTestId,
  feedbackMessage,
  isInvalid,
  isValid,
  label,
  onChange,
  passwordIsHidden = true,
  placeholder,
  value,
  width = 200
}: InputPasswordProps, ref?: LegacyRef<HTMLDivElement>) => {
  const classes = useStyles()
  const [passwordIsHiddenState, setPasswordIsHiddenState] =
    useState<boolean>(passwordIsHidden)

  const containerAttributes = {
    'data-testid': dataTestId,
    ref: ref,
    className: classes.container,
    style: { width: `${width}px` }
  }
  const feedbackIconProps: IconProps = {
    color: isInvalid ? danger : success,
    height: 16,
    icon: isInvalid ? 'danger' : 'success',
    width: 16
  }
  const iconContainerAttributes = {
    className: classes.iconContainer,
    'data-testid': `${dataTestId}IconContainer`,
    onClick: () => setPasswordIsHiddenState(!passwordIsHiddenState),
  }
  const iconProps: IconProps = {
    color: black,
    height: 14,
    icon: passwordIsHiddenState ? 'hide' : 'show',
    width: 18
  }
  const inputAttributes = {
    className: classes.inputElement,
    'data-testid': `${dataTestId}Input`,
    onChange: onChange,
    placeholder: placeholder,
    type: passwordIsHiddenState ? 'password' : 'text',
    value: value
  }
  const inputWrapperAttributes = {
    className: classes.inputWrapper,
    style: isInvalid ? isInvalidStyle : isValid ? isValidStyle : {}
  }
  const spanFeedbackMessageAttributes = {
    className: classes.feedbackMessageContainer,
    'data-testid': `${dataTestId}SpanFeedbackMessage`,
    style: {
      color: isInvalid ? danger : success
    }
  }

  return (
    <div {...containerAttributes}>
      {label && <span className={classes.label}>{label}</span>}
      <span {...inputWrapperAttributes}>
        <input {...inputAttributes} />
        <span {...iconContainerAttributes}>
          <Icon {...iconProps} />
        </span>
      </span>
      {(isInvalid || isValid) && <span {...spanFeedbackMessageAttributes}>
        <Icon {...feedbackIconProps} />
        <span className={classes.feedbackMessageText}>{feedbackMessage}</span>
      </span>}
    </div>
  )
}))

export default InputPassword
