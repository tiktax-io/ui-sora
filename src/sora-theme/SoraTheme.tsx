import React, { memo } from 'react'
import { ThemeProvider } from 'react-jss'
import { SoraThemeProps } from './SoraTheme.types'

/******************************************************************************
 * Wraps children elements and styles them with the theme passed as prop.
 * @param {ReactNode} children JSX elements to be styled with context.
 * @param {Theme} theme object containing the styles.
 * @returns {React.FC} Wrapper with theme containing styles as a Context.
 *****************************************************************************/
const SoraTheme = memo(function SoraTheme({
  children,
  theme
}: SoraThemeProps) {
  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  )
})

export default SoraTheme
