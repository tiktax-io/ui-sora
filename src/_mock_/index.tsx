import { InputDateProps } from '../input-date/InputDate.types'
import { InputPasswordProps } from '../input-password/InputPassword.types'
import { InputSelectProps } from '../input-select/InputSelect.types'
import { InputTextProps } from '../input-text/InputText.types'

export const mockTheme = {
  default_container: {
    background: 'lime',
    border: 'green',
    borderRadius: '4px',
    padding: '20px'
  },
  default_label: {
    color: 'white',
    fontSize: '20px'
  }
}

export const mockInputDateProps: InputDateProps = {
  label: 'Custom label',
  onChange: (newValue: string) => console.log(`New value is: ${newValue}`),
  value: '2025-12-25'
}

export const mockInputPasswordProps: InputPasswordProps = {
  label: 'Custom label',
  onChange: (newValue?: string) => console.log(`New value is: ${newValue}`),
  placeholder: 'Custom placeholder'
}

export const mockInputSelectProps: InputSelectProps = {
  label: 'Custom label',
  onChange: (newValue: string) => console.log(`New value is: ${newValue}`),
  options: [
    'Bitcoin',
    'Ethereum',
    'Bit2me'
  ]
}

export const mockInputTextProps: InputTextProps = {
  label: 'Custom label',
  onChange: (newValue: string) => console.log(`New value is: ${newValue}`),
  placeholder: 'Custom placeholder'
}
