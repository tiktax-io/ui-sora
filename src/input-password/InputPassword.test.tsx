import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import InputPassword from './InputPassword'
import { InputPasswordProps } from './InputPassword.types'
import SoraTheme from '../sora-theme/SoraTheme'
import { mockTheme } from '../_mock_'
import { darkDefaultTheme } from '..'

describe('InputPassword component', () => {

  /****************************************************************************
   * Reset DOM after each test
   ***************************************************************************/
  const container = document.createElement('div')
  beforeEach(() => {
    document.body.appendChild(container)
  })
  afterEach(() => {
    document.body.removeChild(container)
  })

  test('Controlled forms are possible accepting value prop', () => {
    const props: InputPasswordProps = {
      onChange: () => console.log('Lore Ipsum'),
      value: 'Random value'
    }
    render(
      <InputPassword {...props} />
    )
    const demoElement = screen.getByDisplayValue(/Random value/i)
    expect(demoElement).toBeInTheDocument()
  })

  test('Renders feedback message if specified', () => {
    const props: InputPasswordProps = {
      feedbackMessage: 'Custom feedback message',
      isInvalid: true,
      onChange: () => console.log('Lore Ipsum'),
      dataTestId: 'Demo',
      value: 'Random value'
    }
    render(
      <InputPassword {...props} />
    )
    const demoElement = screen.getByTestId(/DemoSpanFeedbackMessage/i)
    expect(demoElement).toBeInTheDocument()
    expect(demoElement).toHaveTextContent(/Custom feedback message/i)
  })

  test('Clicking icon toggles password visibility', () => {
    const props: InputPasswordProps = {
      onChange: () => console.log('Lore Ipsum'),
      dataTestId: 'TestVisibility',
      value: 'MyPassword123'
    }
    render(
      <InputPassword {...props} />
    )
    const toggleIcon = screen.getByTestId(/TestVisibilityIconContainer/i)
    expect(toggleIcon).toBeInTheDocument()
    const revealedPassword = screen.getByTestId(/TestVisibilityInput/i)
    expect(revealedPassword).toHaveAttribute('type', 'password')
    fireEvent.click(toggleIcon)
    expect(revealedPassword).toHaveAttribute('type', 'text')
  })

  /****************************************************************************
   * Testing styles options.
   ***************************************************************************/
  test('Default InputPassword styles matches snapshot', () => {
    const props: InputPasswordProps = {
      onChange: () => console.log('Changed value'),
      value: 'Random value'
    }
    const tree = renderer
      .create(
        <InputPassword {...props}/>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Invalid InputPassword styles matches snapshot', () => {
    const props: InputPasswordProps = {
      feedbackMessage: 'Error feedback message',
      isInvalid: true,
      onChange: () => console.log('Changed value'),
      value: 'Random value'
    }
    const tree = renderer
      .create(
        <InputPassword {...props}/>
      )
      .toJSON()
    expect(tree).toMatchSnapshot() 
  })

  test('InputPassword with successful feedback styles matches snapshot', () => {
    const props: InputPasswordProps = {
      feedbackMessage: 'Optional successful message',
      isValid: true,
      onChange: () => console.log('Changed value'),
      value: 'Random value'
    }
    const tree = renderer
      .create(
        <InputPassword {...props}/>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Styles are customized if wrapped by SoraTheme', () => {
    render(
      <div style={{ alignItems: 'center', display: 'flex', gap: '20px' }}>
        <SoraTheme theme={darkDefaultTheme}>
          <InputPassword dataTestId='customized' />
        </SoraTheme>
        <InputPassword dataTestId='default1' />
        <InputPassword dataTestId='default2' />
      </div>
    )
    const CustomInput = screen.getByTestId('customized').getAttribute('class')
    const Default1 = screen.getByTestId('default1').getAttribute('class')
    const Default2 = screen.getByTestId('default2').getAttribute('class')
    expect(Default1).toMatch(`${Default2}`)
    expect(CustomInput).not.toMatch(`${Default1}`)
    expect(CustomInput).not.toMatch(`${Default2}`)
  })

  test('Customized via SoraTheme matches snapshot', () => {
    const tree = renderer
      .create(
        <SoraTheme theme={darkDefaultTheme}>
          <InputPassword />
        </SoraTheme>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})
