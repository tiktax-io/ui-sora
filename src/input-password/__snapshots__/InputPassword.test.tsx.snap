// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword component Customized via SoraTheme matches snapshot 1`] = `
<div
  className="container-0-2-8"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <span
    className="inputWrapper-0-2-13"
    style={Object {}}
  >
    <input
      className="inputElement-0-2-11"
      data-testid="undefinedInput"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
    />
    <span
      className="iconContainer-0-2-12"
      data-testid="undefinedIconContainer"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={14}
        viewBox="0 0 18 14"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.15068 2.53665C5.362 1.5933 6.91511 0.874986 8.77993 0.874986C10.9893 0.874986 12.7584 1.88232 14.0463 3.07887C15.326 4.26557 16.1818 5.66283 16.5865 6.6636C16.6767 6.87961 16.6767 7.12023 16.5865 7.33624C16.2201 8.21944 15.4845 9.46357 14.4017 10.5737L17.2783 12.8268C17.5627 13.051 17.6146 13.4639 17.3904 13.7483C17.1662 14.0327 16.7533 14.0846 16.4689 13.8604L0.281475 1.17276C-0.00377213 0.949087 -0.0537833 0.536747 0.169804 0.251446C0.393419 -0.0338014 0.805758 -0.0838126 1.09122 0.139775L4.15068 2.53665ZM6.13035 4.08784L8.59946 6.02376C8.71431 5.79134 8.77993 5.52884 8.77993 5.22533C8.77993 4.93549 8.67329 4.64018 8.55025 4.38589C8.62681 4.37768 8.70337 4.35034 8.77993 4.35034C10.2291 4.35034 11.4049 5.52611 11.4049 6.97531C11.4049 7.37453 11.3256 7.72999 11.1615 8.05264L12.2416 8.87842C12.5451 8.32061 12.7174 7.68077 12.7174 6.97531C12.7174 4.82612 10.9537 3.03785 8.77993 3.03785C7.76002 3.03785 6.82761 3.45074 6.13035 4.08784ZM8.77993 13.1249C6.57058 13.1249 4.80146 12.1186 3.51358 10.921C2.23445 9.70966 1.37915 8.31241 0.97228 7.33624C0.882593 7.12023 0.882593 6.87961 0.97228 6.6636C1.23314 6.03743 1.67856 5.22806 2.30199 4.41596L4.88075 6.44758C4.85615 6.62805 4.84247 6.81125 4.84247 6.97531C4.84247 9.17373 6.60613 10.9374 8.77993 10.9374C9.29125 10.9374 9.7807 10.8389 10.2291 10.6612L12.2307 12.2362C11.2381 12.7721 10.0869 13.1249 8.77993 13.1249Z"
          fill="#31393C"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`InputPassword component Default InputPassword styles matches snapshot 1`] = `
<div
  className="container-0-2-1"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <span
    className="inputWrapper-0-2-6"
    style={Object {}}
  >
    <input
      className="inputElement-0-2-4"
      data-testid="undefinedInput"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="Random value"
    />
    <span
      className="iconContainer-0-2-5"
      data-testid="undefinedIconContainer"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={14}
        viewBox="0 0 18 14"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.15068 2.53665C5.362 1.5933 6.91511 0.874986 8.77993 0.874986C10.9893 0.874986 12.7584 1.88232 14.0463 3.07887C15.326 4.26557 16.1818 5.66283 16.5865 6.6636C16.6767 6.87961 16.6767 7.12023 16.5865 7.33624C16.2201 8.21944 15.4845 9.46357 14.4017 10.5737L17.2783 12.8268C17.5627 13.051 17.6146 13.4639 17.3904 13.7483C17.1662 14.0327 16.7533 14.0846 16.4689 13.8604L0.281475 1.17276C-0.00377213 0.949087 -0.0537833 0.536747 0.169804 0.251446C0.393419 -0.0338014 0.805758 -0.0838126 1.09122 0.139775L4.15068 2.53665ZM6.13035 4.08784L8.59946 6.02376C8.71431 5.79134 8.77993 5.52884 8.77993 5.22533C8.77993 4.93549 8.67329 4.64018 8.55025 4.38589C8.62681 4.37768 8.70337 4.35034 8.77993 4.35034C10.2291 4.35034 11.4049 5.52611 11.4049 6.97531C11.4049 7.37453 11.3256 7.72999 11.1615 8.05264L12.2416 8.87842C12.5451 8.32061 12.7174 7.68077 12.7174 6.97531C12.7174 4.82612 10.9537 3.03785 8.77993 3.03785C7.76002 3.03785 6.82761 3.45074 6.13035 4.08784ZM8.77993 13.1249C6.57058 13.1249 4.80146 12.1186 3.51358 10.921C2.23445 9.70966 1.37915 8.31241 0.97228 7.33624C0.882593 7.12023 0.882593 6.87961 0.97228 6.6636C1.23314 6.03743 1.67856 5.22806 2.30199 4.41596L4.88075 6.44758C4.85615 6.62805 4.84247 6.81125 4.84247 6.97531C4.84247 9.17373 6.60613 10.9374 8.77993 10.9374C9.29125 10.9374 9.7807 10.8389 10.2291 10.6612L12.2307 12.2362C11.2381 12.7721 10.0869 13.1249 8.77993 13.1249Z"
          fill="#31393C"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`InputPassword component InputPassword with successful feedback styles matches snapshot 1`] = `
<div
  className="container-0-2-1"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <span
    className="inputWrapper-0-2-6"
    style={
      Object {
        "border": "1px solid #7CEA9C",
        "boxShadow": "0px 4px 10px 3px rgba(124, 234, 156, 0.25)",
      }
    }
  >
    <input
      className="inputElement-0-2-4"
      data-testid="undefinedInput"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="Random value"
    />
    <span
      className="iconContainer-0-2-5"
      data-testid="undefinedIconContainer"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={14}
        viewBox="0 0 18 14"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.15068 2.53665C5.362 1.5933 6.91511 0.874986 8.77993 0.874986C10.9893 0.874986 12.7584 1.88232 14.0463 3.07887C15.326 4.26557 16.1818 5.66283 16.5865 6.6636C16.6767 6.87961 16.6767 7.12023 16.5865 7.33624C16.2201 8.21944 15.4845 9.46357 14.4017 10.5737L17.2783 12.8268C17.5627 13.051 17.6146 13.4639 17.3904 13.7483C17.1662 14.0327 16.7533 14.0846 16.4689 13.8604L0.281475 1.17276C-0.00377213 0.949087 -0.0537833 0.536747 0.169804 0.251446C0.393419 -0.0338014 0.805758 -0.0838126 1.09122 0.139775L4.15068 2.53665ZM6.13035 4.08784L8.59946 6.02376C8.71431 5.79134 8.77993 5.52884 8.77993 5.22533C8.77993 4.93549 8.67329 4.64018 8.55025 4.38589C8.62681 4.37768 8.70337 4.35034 8.77993 4.35034C10.2291 4.35034 11.4049 5.52611 11.4049 6.97531C11.4049 7.37453 11.3256 7.72999 11.1615 8.05264L12.2416 8.87842C12.5451 8.32061 12.7174 7.68077 12.7174 6.97531C12.7174 4.82612 10.9537 3.03785 8.77993 3.03785C7.76002 3.03785 6.82761 3.45074 6.13035 4.08784ZM8.77993 13.1249C6.57058 13.1249 4.80146 12.1186 3.51358 10.921C2.23445 9.70966 1.37915 8.31241 0.97228 7.33624C0.882593 7.12023 0.882593 6.87961 0.97228 6.6636C1.23314 6.03743 1.67856 5.22806 2.30199 4.41596L4.88075 6.44758C4.85615 6.62805 4.84247 6.81125 4.84247 6.97531C4.84247 9.17373 6.60613 10.9374 8.77993 10.9374C9.29125 10.9374 9.7807 10.8389 10.2291 10.6612L12.2307 12.2362C11.2381 12.7721 10.0869 13.1249 8.77993 13.1249Z"
          fill="#31393C"
        />
      </svg>
    </span>
  </span>
  <span
    className="feedbackMessageContainer-0-2-2"
    data-testid="undefinedSpanFeedbackMessage"
    style={
      Object {
        "color": "#7CEA9C",
      }
    }
  >
    <svg
      fill="none"
      height={16}
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 8C0 3.58125 3.58125 0 8 0C12.4187 0 16 3.58125 16 8C16 12.4187 12.4187 16 8 16C3.58125 16 0 12.4187 0 8ZM11.6187 6.61875C11.9594 6.27813 11.9594 5.72187 11.6187 5.38125C11.2781 5.04062 10.7219 5.04062 10.3813 5.38125L7 8.7625L5.61875 7.38125C5.27813 7.04062 4.72187 7.04062 4.38125 7.38125C4.04062 7.72187 4.04062 8.27812 4.38125 8.61875L6.38125 10.6187C6.72187 10.9594 7.27813 10.9594 7.61875 10.6187L11.6187 6.61875Z"
        fill="#7CEA9C"
      />
    </svg>
    <span
      className="feedbackMessageText-0-2-3"
    >
      Optional successful message
    </span>
  </span>
</div>
`;

exports[`InputPassword component Invalid InputPassword styles matches snapshot 1`] = `
<div
  className="container-0-2-1"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <span
    className="inputWrapper-0-2-6"
    style={
      Object {
        "border": "1px solid #F45B69",
        "boxShadow": "0px 4px 10px 3px rgba(244, 91, 105, 0.25)",
      }
    }
  >
    <input
      className="inputElement-0-2-4"
      data-testid="undefinedInput"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="Random value"
    />
    <span
      className="iconContainer-0-2-5"
      data-testid="undefinedIconContainer"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={14}
        viewBox="0 0 18 14"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.15068 2.53665C5.362 1.5933 6.91511 0.874986 8.77993 0.874986C10.9893 0.874986 12.7584 1.88232 14.0463 3.07887C15.326 4.26557 16.1818 5.66283 16.5865 6.6636C16.6767 6.87961 16.6767 7.12023 16.5865 7.33624C16.2201 8.21944 15.4845 9.46357 14.4017 10.5737L17.2783 12.8268C17.5627 13.051 17.6146 13.4639 17.3904 13.7483C17.1662 14.0327 16.7533 14.0846 16.4689 13.8604L0.281475 1.17276C-0.00377213 0.949087 -0.0537833 0.536747 0.169804 0.251446C0.393419 -0.0338014 0.805758 -0.0838126 1.09122 0.139775L4.15068 2.53665ZM6.13035 4.08784L8.59946 6.02376C8.71431 5.79134 8.77993 5.52884 8.77993 5.22533C8.77993 4.93549 8.67329 4.64018 8.55025 4.38589C8.62681 4.37768 8.70337 4.35034 8.77993 4.35034C10.2291 4.35034 11.4049 5.52611 11.4049 6.97531C11.4049 7.37453 11.3256 7.72999 11.1615 8.05264L12.2416 8.87842C12.5451 8.32061 12.7174 7.68077 12.7174 6.97531C12.7174 4.82612 10.9537 3.03785 8.77993 3.03785C7.76002 3.03785 6.82761 3.45074 6.13035 4.08784ZM8.77993 13.1249C6.57058 13.1249 4.80146 12.1186 3.51358 10.921C2.23445 9.70966 1.37915 8.31241 0.97228 7.33624C0.882593 7.12023 0.882593 6.87961 0.97228 6.6636C1.23314 6.03743 1.67856 5.22806 2.30199 4.41596L4.88075 6.44758C4.85615 6.62805 4.84247 6.81125 4.84247 6.97531C4.84247 9.17373 6.60613 10.9374 8.77993 10.9374C9.29125 10.9374 9.7807 10.8389 10.2291 10.6612L12.2307 12.2362C11.2381 12.7721 10.0869 13.1249 8.77993 13.1249Z"
          fill="#31393C"
        />
      </svg>
    </span>
  </span>
  <span
    className="feedbackMessageContainer-0-2-2"
    data-testid="undefinedSpanFeedbackMessage"
    style={
      Object {
        "color": "#F45B69",
      }
    }
  >
    <svg
      fill="none"
      height={16}
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0C3.58125 0 0 3.58125 0 8C0 12.4187 3.58125 16 8 16C12.4187 16 16 12.4187 16 8C16 3.58125 12.4187 0 8 0ZM7.25 4.75C7.25 4.3375 7.5875 4 8 4C8.4125 4 8.75 4.33594 8.75 4.75V8.75C8.75 9.16406 8.41406 9.5 8 9.5C7.58594 9.5 7.25 9.16563 7.25 8.75V4.75ZM8 12.5C7.4575 12.5 7.0175 12.06 7.0175 11.5175C7.0175 10.975 7.45719 10.535 8 10.535C8.54281 10.535 8.9825 10.975 8.9825 11.5175C8.98125 12.0594 8.54375 12.5 8 12.5Z"
        fill="#F45B69"
      />
    </svg>
    <span
      className="feedbackMessageText-0-2-3"
    >
      Error feedback message
    </span>
  </span>
</div>
`;
